src/sql/platform/connection/common/connectionProfileGroup.ts

```diff
@@ -23,6 +23,7 @@ export class ConnectionProfileGroup extends Disposable implements IConnectionPro
 	private _childConnections: ConnectionProfile[] = [];
 	public parentId?: string;
 	private _isRenamed = false;
+	public isRoot: boolean = false;
```

Comment:
It might be useful to just have a function on the service to return the root (which should always be a singleton unless I'm missing something). But I didn't want to mess with this stuff too much so just did enough to get rid of the wonky logic we were using before which used the node name instead.

Reply:
Should this be readonly?